### PR TITLE
Feature/add identicons (and more)

### PR DESCRIPTION
**Pull request by JulianLeviston**
This PR:

* highlights the current player
* adds identicons (in lieu of actual gravatar URLs until we have such things)
* puts the "call liar" button inside the current player's box
* adds some pretty average looking colours
* removes all the message stuff for tableview "component" and makes it a simple function
* Added some config stuff for passing messages into `TablePlayer.view` so it can know how to `callLiar`, and in the future, do other actions.

Would like to eventually use Elm-CSS instead of inline styles: it'd be nice to pretty it up properly.

Need to eventually make it show the current bid of the current player, too.

image

### CODE REVIEW

File: src/app/Table.elm [hunk 1]
```diff
@@ -11,93 +11,114 @@ import Html exposing (..)
...
+
+        optionalLiarButtonDiv =
+            if isCurrentPlayer then
+                [ div [ onClick <| callLiar id ] [ liarButton name ] ]
+            else
+                []
+    in
+        div
+            [ style
+                [ ( "border", "5px solid #262" )
```
Comment (mordrax) [inline src/app/Table.elm]: let's use ElmCSS at some point!

File: src/app/Table.elm [hunk 2]
```diff
@@ -11,93 +11,114 @@ import Html exposing (..)
...
-playerView { id, name, diceCount } =
-    div
-        [ style
-            [ ( "border", "1px solid green" )
-            , ( "width", "200px" )
-            , ( "padding", "10px" )
-            , ( "margin", "8px" )
-            , ( "display", "inline-block" )
+playerView : TableConfig msg -> TablePlayer -> Bool -> Html msg
+playerView { callLiar } { id, name, diceCount } isCurrentPlayer =
```
Comment (mordrax) [inline src/app/Table.elm]: I have a feeling that isCurrentPlayer and lines 79, 85 can be pushed up to the caller so we just have playerView and currentPlayerView which styles players differently with no logic.

Motivation from recent read about removing conditionals or at least pushing it up kinda like side effects.

File: src/app/Table.elm [hunk 3]
```diff
@@ -11,93 +11,114 @@ import Html exposing (..)
...
+                , ( "background-color", backgroundColourOfCurrentPlayer )
+                , ( "padding", "10px" )
+                , ( "margin", "8px" )
+                , ( "display", "inline-block" )
+                ]
             ]
-        ]
-        [ div [] [ text name ]
-        , div [] [ text <| Dice.describeCount diceCount ]
+            ([ div [ iconStyle ] [ identicon "50px" name ]
```
Comment (mordrax) [inline src/app/Table.elm]: cool, it uses the name which is a string as a hashing function? to create a unique icon avatar :)

File: src/app/Table.elm (hunk 1 above)
Comment (JulianLeviston) [inline src/app/Table.elm]: I mentioned that in the doc of this PR.

File: src/app/Table.elm (hunk 3 above)
Comment (JulianLeviston) [inline src/app/Table.elm]: that's right 👍

File: src/app/Table.elm (hunk 2 above)
Comment (JulianLeviston) [inline src/app/Table.elm]: Nah, what about line 86?

Comment (JulianLeviston) [inline src/app/Table.elm]: So you want to pass in two more arguments to this function? Is that what you were thinking?

Comment (JulianLeviston) [inline src/app/Table.elm]: Or, I'd could pass in a new value... `type alias TablePlayerConfig = { backgroundColor : String, buttons : List Html msg }`. ?

File: src/app/Table.elm [hunk 4]
```diff
@@ -11,93 +11,114 @@ import Html exposing (..)
...
+import Identicon exposing (identicon)
 
 
 type alias DiceCount =
     Int
 
 
 type alias TablePlayer =
-    { id : PlayerID
+    { id : Int
```
Comment (bkbooth) [inline src/app/Table.elm]: I probably missed something, but is there a reason we reverted to the less specific `Int` type here (and for the new `currentPlayerId` state property a few lines below)?

**Comment by bkbooth** [comment]
🚢 when you're 😃

File: src/app/Table.elm (hunk 4 above)
Comment (JulianLeviston) [inline src/app/Table.elm]: This was my attempt at having this module have its own private types, but no... it's probably a mistake (I *am* importing `PlayerID`, weird that it changed). I'll adjust. Thanks for the pickup.

**Comment by JulianLeviston** [comment]
Thanks @bkbooth

**Pull request merged**